PostHog/Replay/PostHogDefaultHTTPProtocol.swift

```diff
@@ -0,0 +1,281 @@
+//
+//  PostHogDefaultHTTPProtocol.swift
+//  PostHog
+//
+//  Created by Yiannis Josephides on 07/11/2024.
+//
+
+#if os(iOS)
+    import Foundation
+
+    private let RequestHandledKey = "PHRequestHandled"
+
+    /**
+     URLProtocol is a part of the Foundation framework in iOS, macOS, and other Apple platforms.
+
+     Key Methods:
+       - URLProtocol has 4 key methods
+         - `canInit(with:)`: Called by the system to determine whether the URLProtocol instance should handle a specific request
+         - `canonicalRequest(for:)`: Called right after canInit(with:) returns true and gives us the opportunity to modify the request in any way before feeding it back to the system
+         - `startLoading()`: Called to begin processing the request and do the work needed
+         - `stopLoading()`: Called in the event that the request was canceled
+
+     NOTE: `URLProtocol` implementations need to be registered by calling `URLProtocol.registerClass()` before they can be visible to the URL loading system. If a class is not registered, then `canInit(with:)` will never be called
+
+     NOTE: `URLSessionConfiguration.protocolClasses` - The system calls `canInit(with:)` for each protocol class in the order they are listed here. Therefore, custom protocols are typically inserted at index 0 to ensure higher priority.
+     */
+    final class PostHogHTTPProtocol: URLProtocol {
+        private var session: URLSession?
+        private var sessionDataTask: URLSessionDataTask?
+        private var currentSample = NetworkRequestSample()
+        private var response: URLResponse?
+
+        override init(request: URLRequest, cachedResponse: CachedURLResponse?, client: URLProtocolClient?) {
+            super.init(request: request, cachedResponse: cachedResponse, client: client)
+
+            if session == nil {
+                // ⚠️ - This is currently always using a fresh session with the `default` configuration
+                //    - Need to figure out a way to use custom sessions and configurations here
```

Comment:
That's one of the reasons I gave up on URLProtocol back then since it'd require figuring out more stuff rather than inspiring from the current DD approach.
We need to be sure that React native works as well, I'd need to check if they use a custom or not.